client/src/pages/Signup.js

```diff
@@ -104,39 +57,31 @@ export default function SignUp() {
 
   const validate = () => {
     reset();
-    let isErr = false;
+    let errorMessage = {};
+
     if (email.indexOf("@") === -1 || email === "" || email.length < 5) {
-      setValidError({
-        ...validError,
-        emailErr: "Please enter a valid email with at least 5 characters",
-      });
-      isErr = true;
+      errorMessage.emailErr =
+        "Please enter a valid email with at least 5 characters";
     }
     if (password.length < 5) {
-      setValidError({
-        ...validError,
-        passwordErr: "Password should be at least 8 characters",
-      });
-      isErr = true;
+      errorMessage.passwordErr = "Password should be at least 8 characters";
     }
     if (password !== rePassword && password !== "") {
-      setValidError({ ...validError, retypePassErr: "Passwords do not match" });
-      isErr = true;
+      errorMessage.retypePassErr = "Passwords do not match";
     }
     if (name.length < 3 || name.length > 30) {
-      setValidError({
-        ...validError,
-        nameErr:
-          "Name should be at least 3 characters long. Maximum length of 30 allowed",
-      });
-      isErr = true;
+      errorMessage.nameErr =
+        "Name should be at least 3 characters long. Maximum length of 30 allowed";
     }
-    return isErr;
+
+    setValidError({ ...errorMessage });
+    return errorMessage === {};
```

Comment:
Sign up happens even if passwords don't match. I think this is due to comparing objects.

Reply:
oops😱

Reply:
put it back the `isErr` variable 🙂